Sources/ScrollingTabView.swift

```diff
@@ -339,11 +357,55 @@ public class ScrollingTabViewFlowLayout: UICollectionViewFlowLayout {
                 }
             }
         }
+
+        if let collectionView = self.collectionView
+            where collectionView.contentSize.width < collectionView.frame.width
+                && flexibleWidth {
+            var index = 0
+            var attributesIndex = 0
+
+            let attributesCopy = updatedAttributes
+
+            for attribute in attributesCopy {
+                if attribute.representedElementCategory == .Cell {
+                    if let layoutAttributes = layoutAttributesForItemAtIndexPath(NSIndexPath(forItem: index, inSection: 0)) {
+                        updatedAttributes.removeAtIndex(attributesIndex)
+                        updatedAttributes.insert(layoutAttributes, atIndex: attributesIndex)
+                    }
+                    index++
+                }
+
+                attributesIndex++
+            }
+        }
         
         return updatedAttributes
     }
+
+    override public func layoutAttributesForItemAtIndexPath(indexPath: NSIndexPath) -> UICollectionViewLayoutAttributes? {
+        let attribute = super.layoutAttributesForItemAtIndexPath(indexPath)!
+
+        guard let collectionView = collectionView, let dataSource = collectionView.dataSource
+            where collectionView.contentSize.width < collectionView.frame.width
```

Comment:
Could you ask the delegate for the size of the items from the collection view delegate?